share/smack/lib/half_intrinsics.c

```diff
@@ -0,0 +1,29 @@
+#include "smack.h"
+
+// Now this guard is unnecessary but if we decide to use this file
+// to implement all the intrinsics, we will need it.
+#if FLOAT_ENABLED
+float __SMACK_convert_from_fp16_f32(unsigned short x) {
+  float ret = __VERIFIER_nondet_float();
+  __SMACK_code("@f := $fpext.bvhalf.bvfloat(RNE, $bitcast.bv16.bvhalf(@H));", ret, x);
```

Comment:
Should rounding mode be hard-coded here?

Reply:
All the lower precision floating-point representations can be precisely represented by higher precision ones. The rounding mode doesn't matter since no rounding would occur.

Reply:
While no rounding is involved in these upcasts, it shouldn't hurt to be consistent and use the current rounding mode for all conversions.

Reply:
Yes, I think what @keram88 says here is a good idea. Let's use the current rounding mode everywhere.